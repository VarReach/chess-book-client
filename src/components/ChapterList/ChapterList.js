import React, { Component } from 'react';
import ChapterPreview from '../ChapterPreview/ChapterPreview';
import UserContext from '../../contexts/UserContext';
import helpers from '../../helpers/misc-helpers';

class CompletedPreview extends Component {
  static contextType = UserContext;

  getCompletedChapterIds = (chapterId, completedChapters, completed_chapters) => {
    if (completedChapters && chapterId) {
      const index = completedChapters.indexOf(chapterId);
      if (index !== -1) {
        let date = completed_chapters[index].date_completed;
        date = helpers.parseDate(date);
        return date;
      }
    }
  }

  renderChapters = () => {
    let completedChapters;
    let completed_chapters;
    if (this.context && this.context.user) {
      completed_chapters = this.context.user.completed_chapters;
      completedChapters = completed_chapters ? completed_chapters.map(cc => cc.id) : null;
    }
    return this.props.chapters.map((chapter, i) => {
      const date = this.getCompletedChapterIds(chapter.id, completedChapters, completed_chapters);
      return (
        <ChapterPreview key={i} index={i} chapter={chapter} date={date}/>
      )
    });
  }

  render() {
    return (
      <section className="container">
        {this.props.chapters && this.renderChapters()}
      </section>
    )
  }
}

export default CompletedPreview;