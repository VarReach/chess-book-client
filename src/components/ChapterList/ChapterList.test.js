import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChapterList from './ChapterList';

describe('ChapterList component', () => {
  const props = {
    chapters: [
      {
        title: 'chapter 1 title'
      },
      {
        title: 'chapter 2 title'
      }
    ],
  }

  it('renders the list', () => {
    const wrapper = shallow(<ChapterList/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the list given props', () => {
    const wrapper = shallow(<ChapterList {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});