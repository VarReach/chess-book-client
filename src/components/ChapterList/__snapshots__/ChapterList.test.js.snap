// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterList component renders the list 1`] = `
<section
  className="container"
/>
`;

exports[`ChapterList component renders the list given props 1`] = `
<section
  className="container"
>
  <ChapterPreview
    chapter={
      Object {
        "title": "chapter 1 title",
      }
    }
    index={0}
    key="0"
  />
  <ChapterPreview
    chapter={
      Object {
        "title": "chapter 2 title",
      }
    }
    index={1}
    key="1"
  />
</section>
`;
